### ISSUE TITLE
Reasonable values for loss and number of epochs

### ISSUE THREAD

**Issue by rsatapat** [alzayats/DeepFish#18]
Hi,

I am trying to train the dataset for the localisation task. Since we don't have a particularly beefy GPU, each epoch is taking quite some time (around 15 mins). Could you suggest some reasonable values for loss and the minimum number of epochs you think are needed to get good results?

Thanks,
Roshan

**Comment by alzayats** [comment]
Hi Roshan,

For localization, we trained for 1,000 epochs with batch size of 1. The pretrained model achieved MAE of 0.38 (meaning ~0.38 miscounts per image on average). https://www.nature.com/articles/s41598-020-71639-x 

With 15 min/epoch, 1,000 epochs isn't realistic for you. I'd suggest starting with 100-200 epochs and monitoring validation MAE - if it plateaus, you can stop.

Also, you might want to try free GPU options like Google Colab or Kaggle notebooks - they should speed things up significantly.

Cheers,
Alzayat Saleh

**Comment by rsatapat** [comment]
Hi, Thanks for your reply.

I have one followup question. The training script generates a file called score_list.pkl. Is my understanding correct that the variable "val_loc" for each epoch is the validation MAE you are referring to?
I have attached a plot with values for val_loc and loss_loc for the first 70 trials. 

<img width="547" height="413" alt="Image" src="https://github.com/user-attachments/assets/5885efa5-a7fb-4b43-9a2d-cc42f1d84049" />

**Comment by alzayats** [comment]
Hi Roshan,

Yes, val_loc is the validation MAE. Your curves look good - val_loc is around 0.2-0.3 at epoch 70.

The curves are still dropping slightly, so you could continue for another 30-50 epochs, but honestly you're already in a good range. If val_loc stays flat for 10-15 epochs, you're done.

**Comment by rsatapat** [comment]
Perfect! Thanks you so much.
